upgrade/kind.go

```diff
@@ -185,9 +185,22 @@ var getRepoKind = stepv2.Func11E("Get Repo Kind", func(ctx context.Context, repo
 		Bridge:   bridge,
 		Pf:       pf,
 	}
-	// get the org name that hosts the upstream repo
-	tok := strings.Split(modPathWithoutVersion(upstream.Mod.Path), "/")
-	out.UpstreamProviderOrg = tok[len(tok)-2]
+	// get the org name that hosts the upstream repo from the GitHubOrg field in provider/resources.go
+	resourceFile := filepath.Join(path, "provider", "resources.go")
+	resourceData := stepv2.ReadFile(ctx, resourceFile)
+	resLines := strings.Split(resourceData, "\n")
+	var upstreamOrg string
+	for _, line := range resLines {
+		if strings.Contains(line, "GitHubOrg") {
+			lineSlice := strings.Split(line, "\"")
+			upstreamOrg = lineSlice[1]
+			break
+		}
+	}
+	if upstreamOrg == "" {
+		contract.Assertf(upstreamOrg != "nil", "upstreamOrg cannot be empty; set GitHubOrg in ProviderInfo")
```

Comment:
Eh,  this really not robust at all and will cause problems, reading source code .. there's no guarantee it's a literal or anything. I know this can be a pain to extract but we need to file a ticket to get the right. Also if we're going ahead with these approximate methods how about falling back gently? Fall back on the previously working method if not detected robustly by this new method? 


This field is explicitly optional in the bridge.

```
func (info ProviderInfo) GetGitHubOrg() string {
	if info.GitHubOrg == "" {
		return "terraform-providers"
	}

	return info.GitHubOrg
}

```

Reply:
The "right"-er way would be either this moves to a config file or if it's in Go source then we compile the go file and call the Go software to self-describe.

Reply:
Yeah, the options are not great.

We can of course fall back on the previous implementation if we cannot find an upstream org scanning the resources.go file.

But that doesn't solve for the case where `GitHubOrg` is set incorrectly (nor would compiling the source code and querying ProviderInfo). 
Submodule detection does not work either, for the same reason this is a bug in the first place.

These are all reasonable arguments for using an upgrade config but I was hoping we could avoid manual input. :/